index.bs

```diff
@@ -290,41 +296,49 @@ spec: HEADER-STRUCTURE; urlPrefix: https://httpwg.org/http-extensions/draft-ietf
-        </ol>
-      </li>
-      <li>return false.</li>
-    </ol>
+    <div algorithm=matches>
+      <p>To determine whether an <a>allowlist</a> <dfn>matches</dfn> an origin
+      <var>origin</var>, run these steps:
+
+      1. If the <a>allowlist</a> is <a>the special value <code>*</code></a>,
+         then return true.
+
+      1. Otherwise, [=set/for each=] |item| in the <a>allowlist</a>:
+
+        1. If |item| is [=same origin-domain=] with <var>origin</var>, then
+           return true.
+
+      1. Return false.
+    </div>
   </section>
   <section>
     <h3 id="default-allowlists">Default Allowlists</h3>
     <p>Every <a>policy-controlled feature</a> has a <dfn export
     lt="default allowlist|default allowlists"
     for="policy-controlled feature">default allowlist</dfn>. The <a>default
-    allowlist</a> determines whether the feature is allowed in a document with
-    no declared policy in a [=/top-level traversable=], and also whether access
-    to the feature is automatically delegated to documents in [=child
-    navigables=].</p>
+    allowlist</a> determines whether the feature is allowed in a {{Document}} with
+    no <a for="permissions policy">declared policy</a> in a [=/top-level
+    traversable=], and also whether access to the feature is automatically
+    delegated to documents in [=child navigables=].</p>
     <p>The <a>default allowlist</a> for a <a
     data-lt="policy-controlled feature">feature</a> is one of these values:</p>
     <dl>
       <dt><dfn for="default allowlist" export><code>*</code></dfn></dt>
-      <dd>The feature is allowed in documents in top-level traversables by
-      default, and when allowed, is allowed by default to documents in child
-      navigables.</dd>
+      <dd>The feature is allowed in {{Document}}s in [=/top-level
+      traversables=] by default, as well as those in all [=child navigables=].
+      It can be disallowed in [=child navigables=] by explicitly supplying a
+      [=container policy=] on the [=navigable container=] that overrides this
+      default.</dd>
       <dt><dfn for="default allowlist" export><code>'self'</code></dfn></dt>
-      <dd>The feature is allowed in documents in top-level traversables by
-      default, and when allowed, is allowed by default to same-origin domain
-      documents in child navigables, but is disallowed by default in
-      cross-origin documents in child navigables.</dd>
+      <dd>The feature is allowed in [=navigable/active document|documents=] in
+      [=/top-level traversables=] by default, as well as those in [=child
+      navigables=] whose [=navigable/active document|document=] is
+      [=same origin=] with the [=/top-level traversable=]'s [=navigable/active
+      document|document=]. It is disallowed by default in [=child navigables=]
+      whose [=navigable/active document|document=] is cross-origin with the
+      [=/top-level traversable=]'s [=navigable/active document|document=], but
```

Comment:
Is this true? It can't always be enabled in a child navigable via the `Permisisons-Policy` header inside. The only thing that header could do *for sure* (without being dependent on the container policy) is _disable_ the feature, which I've clarified above. WDYT?

Reply:
> disallowed by default in child navigables whose document is cross-origin with the top-level traversable's document

The point here is that we don't check, for a given non-top-level frame, whether it's cross-origin with the top-level; we check its relationship with it's parent frame. If it's cross-origin with it's parent frame, then a default-self feature will be disabled in the child unless the parent frame `allow`s it. (It's not a sufficient condition; there are other conditions that need to be met as well, like the feature actually being available to the parent frame, and the child frame not declining the feature through a `permissions-policy` header). Regardless of whether it's same-origin with the top-level frame, default-self features are disabled across origin boundaries by default.